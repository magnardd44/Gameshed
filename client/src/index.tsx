import * as React from 'react';
import { Component } from 'react-simplified';
import ReactDOM from 'react-dom';
import {
  Card,
  Alert,
  Row,
  Form,
  Column,
  Button,
  NavBar,
  Header,
  Container,
  ColumnCentre,
  SearchResult,
} from './widgets';
import axios from 'axios';
import { HashRouter, Route } from 'react-router-dom';
import { AddReview, PublishReview } from './review-components';
import { Hash } from 'crypto';
import { createHashHistory } from 'history';

axios.defaults.baseURL = 'http://localhost:3000/api/v2';

const history = createHashHistory();

class NavHeader extends Component {
  render() {
    return (
      <div>
        <NavBar brand="GameShed">
          <NavBar.Link to="/games">Games</NavBar.Link>
        </NavBar>
        <Header></Header>
      </div>
    );
  }
}

class Search extends Component {
  input = '';
  render() {
    return (
      <>
        <Container textalign={'centre'}>
          <Row>
            <ColumnCentre width={6} offset={2}>
              <Form.Input
                type={this.input}
                value={this.input}
                placeholder="Søk etter et spill"
                onChange={(event) => (this.input = event.currentTarget.value)}
              />
            </ColumnCentre>
            <ColumnCentre width={2}>
              <Button.Success
                onClick={() => {
                  this.search();
                }}
              >
                Søk
              </Button.Success>
            </ColumnCentre>
          </Row>
        </Container>
      </>
    );
  }
  search() {
    console.log('search');
    history.push('/results');
  }
}

class SearchListings extends Component {
  render() {
    return (
      <Container>
        Søkeresultater:
        <SearchResult></SearchResult>
        <SearchResult></SearchResult>
        <SearchResult></SearchResult>
      </Container>
    );
  }
}

ReactDOM.render(
  <HashRouter>
    <div>
      <NavHeader />
      <Route exact path="/" component={Search}></Route>
      <Route exact path="/results" component={SearchListings}></Route>
      <Route exact path="/addReview" component={AddReview} />
      <Route exact path="/publishReview/:id" component={PublishReview} />
    </div>
  </HashRouter>,
  document.getElementById('root')
);

/*
class Start extends Component {
  input = '';
  stdout = '';
  stderr = '';
  errCode: number | null = null;
}



  render() {
    return (
      <>
        <Card title="GameShed">
          <Row>
            <Column>
              <Form.Label>Søk:</Form.Label>
              <Form.Textarea
                value={this.input}
                onChange={(event) => (this.input = event.currentTarget.value)}
              />
            </Column>
          </Row>
          <Button.Success
            onClick={() => {
              axios
                .post<{ exitStatus: number; stdout: string; stderr: string }>('/run', {
                  language: 'js',
                  source: this.input,
                })
                .then((response) => {
                  this.errCode = response.data.exitStatus;
                  this.stdout = response.data.stdout;
                  this.stderr = response.data.stderr;
                })
                .catch((error: Error) => Alert.danger('Could not run app.js: ' + error.message));
            }}
          >
            Søk etter spill
          </Button.Success>
        </Card>
        <Card title="Standard output">{this.stdout}</Card>
        <Card title="Standard error">{this.stderr}</Card>
        <Card title="Errorcode">{this.errCode}</Card>
      </>
    );
  }
}


const root = document.getElementById('root');
if (root)
  ReactDOM.render(
    <HashRouter>
      <Alert />

      <Route exact path="/" component={Start} />
      
    </HashRouter>,
    root
  );
 
 */
