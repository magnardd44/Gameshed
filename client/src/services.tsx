import axios from 'axios';
import { sharedComponentData } from 'react-simplified';

axios.defaults.baseURL = 'http://localhost:3000/api/v2';

export type Game = {
  game_id: number;
  game_title: string;
  genre: string[];
  genre_id: number;
  platform: string[];
  game_description: string;
};

export type Review = {
  review_id: number;
  game_id: string;
  external_game_id: string[];
  game_title: number;
  genre: string[];
  platform: string;
  review_title: string;
  text: string;
  user_id: number;
  rating: number;
  published: boolean;
};

class GameService {
  game: Game = {
    game_id: 0,
    game_title: '',
    genre: [],
    genre_id: 0,
    platform: [],
    game_description: '',
  };
  games: Game[] = [];

  /**
   * Get game with given id.
   */
  get(id: number) {
    return axios.get<Game>('/games/' + id).then((response) => response.data);
  }

  /**
   * Get all games.
   */
  getAll() {
    return axios.get<Game[]>('/games').then((response) => response.data);
  }

  /**
   * Create new task having the given title.
   *
   * Resolves the newly created task id.
   */
  create(title: string, genre: string, platform: number, description: string) {
    return axios
      .post<{ id: number }>('/games', {
        game_title: title,
        genre: genre,
        platform: platform,
        game_description: description,
      })
      .then((response) => response.data.id);
  }
}

class ReviewService {
  review: Review = {
    review_id: 0,
    game_id: '',
    external_game_id: [],
    game_title: 0,
    genre: [],
    platform: '',
    review_title: '',
    text: '',
    user_id: 0,
    rating: 0,
    published: false,
  };
  reviews: Review[] = [];

  /**
   * Get review with given id.
   */
  get(id: number) {
    return axios.get<Review>('/reviews/' + id).then((response) => response.data);
  }

  /**
   * Get all reviews.
   */
  getAll() {
    return axios.get<Review[]>('/reviews').then((response) => response.data);
  }

  /**
   * Create new review having the given title.
   *
   * Resolves the newly created review id.
   */
  create(title: string, text: string, rating: number) {
    return axios
      .post<{ id: number }>('/reviews', { review_title: title, text: text, rating: rating })
      .then((response) => response.data.id);
  }
}

export let gameService = sharedComponentData(new GameService());
export let reviewService = sharedComponentData(new ReviewService());
