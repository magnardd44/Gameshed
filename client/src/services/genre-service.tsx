import axios from 'axios';
import { sharedComponentData } from 'react-simplified';

axios.defaults.baseURL = 'http://localhost:3000/api/v2';

export type Genre = {
  genre_id: number;
  genre_name: string;
};

class GenreService {
  genre: Genre = {
    genre_id: 0,
    genre_name: '',
  };
  genres: Genre[] = [];

  /**
   * Get genre with given id.
   */
  get(id: number) {
    return axios.get<Genre>('/genres/' + id).then((response) => response.data);
  }

  /**
   * Get genre id with given name.
   */
  getId(name: string) {
    return axios.get<Genre>('/genres/' + name).then((response) => response.data);
  }

  /**
   * Get all genres.
   */
  getAll() {
    return axios.get<Genre[]>('/genres').then((response) => response.data);
  }

  /**
   * Create new genre having the given name.
   *
   * Resolves the newly created genre id.
   */
  create(name: string) {
    return axios
      .post<{ id: number }>('/genres', {
        name: name,
      })
      .then((response) => response.data.id);
  }
  updateGenreMap(game_id: number, genre_id: number) {
    return axios
      .post<{ id: number }>('/genreMap', {
        game_id: game_id,
        genre_id: genre_id,
      })
      .then((response) => response.data.id);
  }
}

export let genreService = sharedComponentData(new GenreService());
