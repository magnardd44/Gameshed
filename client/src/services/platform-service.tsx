import axios from 'axios';
import { sharedComponentData } from 'react-simplified';
import userService from './user-service';

axios.defaults.baseURL = 'http://localhost:3000/api/v2';

export type Platform = {
  platform_id: number;
  platform_name: string;
};

class PlatformService {
  platform: Platform = {
    platform_id: 0,
    platform_name: '',
  };
  platforms: Platform[] = [];

  /**
   * Get platform id with given name.
   */
  getId(name: string) {
    return axios.get<Platform>('/platforms/' + name).then((response) => response.data);
  }

  /**
   * Get all platforms.
   */
  getAll() {
    return axios.get<Platform[]>('/platforms').then((response) => response.data);
  }

  /**
   * Create new platform having the given name.
   *
   * Resolves the newly created platform id.
   */
  create(name: string) {
    return userService.axios
      .post<{ id: number }>('/platforms', {
        name: name,
      })
      .then((response) => response.data.id);
  }

  /**
   * Update mappin_platform with platform_id and game_id.
   *
   * Resolves the newly created platform id.
   */
  updatePlatformMap(platform_id: number, game_id: number) {
    return userService.axios.post<{ id: number }>('/platforms/map', {
      platform_id: platform_id,
      game_id: game_id,
    });
  }

  stringToId(genre: string) {
    return this.platforms.find((s) => s.platform_name == genre)?.platform_id || 0;
  }

  idToString(id: number) {
    return this.platforms.find((s) => s.platform_id == id)?.platform_name || 'None';
  }

  updatePlatformMapString(platform: string, game_id: number) {
    return this.updatePlatformMap(this.stringToId(platform), game_id);
  }
}

export let platformService = sharedComponentData(new PlatformService());
