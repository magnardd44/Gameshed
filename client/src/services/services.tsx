import axios from 'axios';
import { sharedComponentData } from 'react-simplified';

axios.defaults.baseURL = 'http://localhost:3000/api/v2';

export type Game = {
  game_id: number;
  game_title: string;
  genre: string[];
  genre_id: number;
  platform: string[];
  game_description: string;
};

class GameService {
  game: Game = {
    game_id: 0,
    game_title: '',
    genre: [],
    genre_id: 0,
    platform: [],
    game_description: '',
  };
  games: Game[] = [];

  /**
   * Get game with given id.
   */
  get(id: number) {
    return axios.get<Game>('/games/' + id).then((response) => response.data);
  }

  /**
   * Get all games.
   */
  getAll() {
    return axios.get<Game[]>('/games').then((response) => response.data);
  }

  /**
   * Create new task having the given title.
   *
   * Resolves the newly created task id.
   */
  create(title: string, genre: string, platform: number, description: string) {
    return axios
      .post<{ id: number }>('/games', {
        game_title: title,
        genre: genre,
        platform: platform,
        game_description: description,
      })
      .then((response) => response.data.id);
  }
}

export let gameService = sharedComponentData(new GameService());
