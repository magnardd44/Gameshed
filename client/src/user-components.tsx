import * as React from 'react';
import { Component } from 'react-simplified';
import { Form, Card, Alert, Button, Container, Column } from './widgets';
import userService from './services/user-service';
//import { history } from './index';
import { createHashHistory } from 'history';
//
const history = createHashHistory();

export class UserNav extends Component {
  //password: string = '';

  render() {
    return (
      <>
        <Button.Success
          small
          onClick={() => {
            userService.login('admin', 'admin').then(() => Alert.success(<>Logged inn as admin</>));
            //this.password = '';
          }}
        >
          DebugAdmin
        </Button.Success>
        <Form.Label>Brukernavn:</Form.Label>
        <Form.Input
          id="emailInput"
          type="text"
          value={userService.email}
          placeholder="Epost"
          disabled={userService.token}
          onChange={(event) => {
            userService.email = event.currentTarget.value;
          }}
        />
        {userService.token ? (
          // If user logged in
          <Button.Success
            small
            onClick={() => {
              userService.logout();
              userService.email = '';
              //this.password = '';
            }}
          >
            Logout
          </Button.Success>
        ) : (
          // else user not logged
          //          <>
          //		  {
          //            //<Form.Input
          //            //  type="text"
          //            //  value={this.password}
          //            //  placeholder="Passord"
          //            //  onChange={(event) => {
          //            //    this.password = event.currentTarget.value;
          //            //  }}
          //            ///>
          //			}
          <Button.Success
            small
            onClick={() => {
              //console.log(userService.email.length ? true : false);
              let password = prompt('Skriv inn passord');
              if (userService.email.length && password?.length) {
                userService
                  .login(userService.email, password)
                  .catch((err) => Alert.warning(<>Feil brukarnavn eller passord</>));
              } else {
                Alert.info(<>Skriv inn brukarnavn og passord</>);
              }
              //                this.password = '';
            }}
          >
            Login
          </Button.Success>
          //</>
        )}
        <Button.Success
          small
          onClick={() => {
            history.push('/user');
            //if(history.location.pathname != '/user')  history.push('/user')
            //                this.password = '';
            //                userService.register(this.input, this.password)
            //				.then(()=> history.push('/user'))
            //				.catch(()=> history.push('/user')
            //				);
            //				:
          }}
        >
          {userService.token ? 'MinSide' : 'Registrer'}
        </Button.Success>
      </>
    );
  }
}

export class UserData extends Component {
  render() {
    return (
      <Card title="Brukerdata">
        <Form.Label>
          Brukarnavn
          <Form.Input
            id="inputName"
            type="text"
            value={userService.name}
            placeholder="Brukernavn"
            onChange={(event) => {
              userService.name = event.currentTarget.value;
            }}
          />
        </Form.Label>

        <Form.Label>
          Epost
          <Form.Input
            id="inputEmail"
            type="text"
            value={userService.email}
            disabled={userService.token}
            placeholder="Epost"
            required={true}
            onChange={(event) => {
              userService.email = event.currentTarget.value;
            }}
          />
        </Form.Label>

        <Form.Label>
          Om meg
          <Form.Textarea
            id="inputAbout"
            type="text"
            value={userService.about}
            placeholder="Om meg"
            onChange={(event) => {
              userService.about = event.currentTarget.value;
            }}
          />
        </Form.Label>
      </Card>
    );
  }
}

export class UserPersonal extends Component {
  render() {
    return (
      <>
        <Button.Success
          onClick={() => {
            userService.set_user().then(() => Alert.success(<>Oppdatert bruker</>));
          }}
        >
          Oppdater bruker
        </Button.Success>
        <Button.Danger
          onClick={() => {
            userService.delete();
            // this.input = '';
            // this.password = '';
          }}
        >
          Slett meg
        </Button.Danger>
      </>
    );
  }
  mounted() {
    let result = userService
      .get_user()
      .then()
      .catch((err) => {
        //console.log('Failed');
        //console.log(err);
      });
  }
}

export class UserRegister extends Component {
  render() {
    return (
      <>
        <Button.Success
          onClick={() => {
            let newPassword = prompt('Skriv inn passord');
            if (newPassword?.length) {
              userService
                .register(userService.email, newPassword)
                .then(() => Alert.success(<>Ny bruker registrert</>));
            } else {
              Alert.info(<>Skriv inn passord</>);
            }
          }}
        >
          Registrer ny bruker
        </Button.Success>
      </>
    );
  }
}

export class UserPage extends Component {
  render() {
    return (
      <>
        <UserData />
        {userService.token ? <UserPersonal /> : <UserRegister />}
      </>
    );
  }
}
