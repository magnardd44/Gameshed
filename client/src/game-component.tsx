import * as React from 'react';
import { Component } from 'react-simplified';
import {
  Alert,
  Card,
  Row,
  Column,
  Form,
  Button,
  ThumbNail,
  Container,
  ColumnCentre,
  Linebreak,
} from './widgets';
import { NavLink } from 'react-router-dom';
import { gameService, reviewService, Game, Review } from './services';
import { createHashHistory } from 'history';

const history = createHashHistory(); // Use history.push(...) to programmatically change path, for instance after successfully saving a student

export class GameCard extends Component {
  game: Game = {
    game_id: 0,
    game_title: '',
    genre: [],
    genre_id: 0,
    platform: [],
    game_description: '',
  };
  description =
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco...';
  render() {
    return (
      <Card title={this.game.game_title}>
        <h6 className="card-subtitle mb-2 text-muted">
          Terningkast:
          <ThumbNail small img="https://cdn-icons-png.flaticon.com/512/220/220725.png"></ThumbNail>
        </h6>
        <Row>
          <Column width={2}>
            <ThumbNail img="https://cdn-icons-png.flaticon.com/512/686/686589.png"></ThumbNail>
          </Column>
          <Column width={6}>
            {this.game.game_description}
            <Linebreak></Linebreak>
          </Column>
          <Column width={2}></Column>
        </Row>
        <Linebreak></Linebreak>
        <Row>
          <Column>Sjanger: {this.game.genre}</Column>
        </Row>
        <Linebreak></Linebreak>
        <Row>
          <Column>Platform: {this.game.platform}</Column>
        </Row>
      </Card>
    );
  }
  /** return axios
   */
  mounted() {
    gameService.getAll().then((result) => {
      this.game = result[0];
      console.log(this.game);
    });
  }
}
