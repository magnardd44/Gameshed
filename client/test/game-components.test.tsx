import * as React from 'react';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import { UserNav, UserData, UserPersonal, UserRegister, UserPage } from '../src/user-components';
import { Form, Card, Alert, Button, Container, Column } from '../src/widgets';
import { shallow } from 'enzyme';
import userService, { Token } from '../src/services/user-service';
import { gameService } from '../src/services/game-service';
import { AddGame } from '../src/game-component';
import { genreService } from '../src/services/genre-service';
import { platformService } from '../src/services/platform-service';

jest.setTimeout(1000);

//const mockGenreAdapter = new MockAdapter(genreService.axios);

//const mockPlatformAdapter = new MockAdapter(platformService.axios);

//const mockUserAdapter = new MockAdapter(userService.axios);

//const mockGlobalAdapter = new MockAdapter(axios);

const mockAdapter = new MockAdapter(axios);

mockAdapter.onGet('/games/1').reply(200, {
  game_id: 1,
  igdb_id: 2,
  game_title: 'test',
  genre: ['a', 'b', 'c'],
  platform: ['a', 'b', 'c'],
  game_description: 'test',
  igdb: null,
});

mockAdapter.onGet('/games').reply(200, [
  {
    game_id: 1,
    igdb_id: 2,
    game_title: 'test',
    genre: ['a', 'b', 'c'],
    platform: ['a', 'b', 'c'],
    game_description: 'test',
    igdb: null,
  },
  {
    game_id: 2,
    igdb_id: 3,
    game_title: 'test2',
    genre: ['a', 'b', 'c'],
    platform: ['a', 'b', 'c'],
    game_description: 'test',
    igdb: null,
  },
]);

mockAdapter.onGet('/genres/1').reply(200, {
  genre_id: 1,
  genre_name: 'test',
  genre_img: '',
});

mockAdapter.onGet('/genres').reply(200, [
  {
    genre_id: 1,
    genre_name: 'test',
    genre_img: '',
  },
  {
    genre_id: 2,
    genre_name: 'test2',
    genre_img: '',
  },
]);

mockAdapter.onGet('/platforms/1').reply(200, {
  platform_id: 1,
  platform_name: 'test',
});

mockAdapter.onGet('/platforms').reply(200, [
  {
    genre_id: 1,
    genre_name: 'test',
    genre_img: '',
  },
  {
    genre_id: 2,
    genre_name: 'test2',
    genre_img: '',
  },
]);

//const mockGlobalAdapter = new MockAdapter(axios);
//const mockGameService = new MockAdapter(gameService.axios);
//const mockGenreService = new MockAdapter(genreService.axios);
//const mockPlatformService = new MockAdapter(platformService.axios);

/*beforeEach(() => {});*/

//mockGlobalAdapter.onGet('user').reply(200, { nick: 'abc', about: 'def', email: 'ghi' });

//userService.token = { id: 1, token: 'test' };

/*
window.prompt = jest.fn(() => {
  return 'password';
});*/

Alert.info = jest.fn(() => {});
Alert.warning = jest.fn(() => {});
Alert.success = jest.fn(() => {});

describe('AddGame component', () => {
  test('AddGame default', () => {
    const wrapper = shallow(<AddGame />);

    // Wait for events to complete
    setTimeout(() => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  test('AddGame saves correctly', (done) => {
    const wrapper = shallow(<AddGame />);

    wrapper.find('#titleInput').simulate('change', { currentTarget: { value: 'test' } });
    wrapper.find('#descriptionInput').simulate('change', { currentTarget: { value: 'test' } });
    wrapper.find('#genreSel').simulate('change', { currentTarget: { value: 'Puzzle' } });
    wrapper.find('#platformSel').simulate('change', { currentTarget: { value: 'Mac' } });

    // @ts-ignore
    wrapper.find(<Button.Success>Lagre</Button.Success>).simulate('click');

    // Wait for events to complete
    //
    setTimeout(() => {
      expect(Alert.success).toBeCalled();
      done();
    });
  });

  test.skip('UserNav on logged out', (done) => {
    const wrapper = shallow(<UserNav />);

    mockAdapter.onPost('user/login').reply(200, { id: 1, token: 'abc' });
    mockAdapter.onPost('user/logout').reply(200);

    wrapper.find('#emailInput').simulate('change', { currentTarget: { value: 'email' } });
    wrapper.find({ children: 'Login' }).simulate('click');

    setTimeout(() => {
      expect(
        wrapper.containsMatchingElement(
          // @ts-ignore
          <Button.Success>Logout</Button.Success>
        )
      ).toEqual(true);
      wrapper.find({ children: 'Logout' }).simulate('click');
      setTimeout(() => {
        expect(
          wrapper.containsMatchingElement(
            // @ts-ignore
            <Button.Success>MinSide</Button.Success>
          )
        ).toEqual(false);
        expect(
          wrapper.containsMatchingElement(
            // @ts-ignore
            <Button.Success>Registrer</Button.Success>
          )
        ).toEqual(true);
        done();
      });
    });
  });

  test.skip('Ingen epost', (done) => {
    const wrapper = shallow(<UserNav />);

    wrapper.find({ children: 'Login' }).simulate('click');

    setTimeout(() => {
      expect(Alert.info).toBeCalled();
      done();
    });
  });

  test.skip('Feil brukar eller passord', (done) => {
    const wrapper = shallow(<UserNav />);

    wrapper.find('#emailInput').simulate('change', { currentTarget: { value: 'email' } });
    mockAdapter.onPost('user/login').reply(400);
    wrapper.find({ children: 'Login' }).simulate('click');

    setTimeout(() => {
      expect(Alert.warning).toBeCalled();
      done();
    });
  });
});

describe.skip('UserData component', () => {
  test('Inputs', (done) => {
    const wrapper = shallow(<UserData />);

    userService.token = { id: 0, token: 'token' };

    setTimeout(() => {
      wrapper.find('#inputName').simulate('change', { currentTarget: { value: 'abc' } });
      wrapper.find('#inputEmail').simulate('change', { currentTarget: { value: 'def' } });
      wrapper.find('#inputAbout').simulate('change', { currentTarget: { value: 'ghi' } });
      setTimeout(() => {
        expect(userService.name).toBe('abc');
        expect(userService.email).toBe('def');
        expect(userService.about).toBe('ghi');
        done();
      });
    });
  });
});

describe.skip('UserRegister component', () => {
  test('Register new ', (done) => {
    userService.email = 'email';
    mockAdapter.onPost('user/add').reply(200, { id: 0, token: 'token' });
    mockAdapter.onGet('user').reply(200, { nick: 'abc', about: 'def', email: 'ghi' });

    const wrapper = shallow(<UserRegister />);

    wrapper.find({ children: 'Registrer ny bruker' }).simulate('click');
    setTimeout(() => {
      expect(userService.name).toEqual('abc');
      done();
    });
  });

  test('Empty password', (done) => {
    window.prompt = jest.fn(() => {
      return '';
    });

    const wrapper = shallow(<UserRegister />);

    wrapper.find({ children: 'Registrer ny bruker' }).simulate('click');
    setTimeout(() => {
      expect(Alert.info).toBeCalled();
      done();
    });
  });
});

describe.skip('UserPersonal component', () => {
  beforeEach(() => {
    userService.token = { id: 0, token: '' };
    mockAdapter.onGet('user').reply(200, { nick: 'abc', about: 'def', email: 'ghi' });
  });

  test('Set user', (done) => {
    const wrapper = shallow(<UserPersonal />);
    mockAdapter.onPut('user').reply(200);

    wrapper.find({ children: 'Oppdater bruker' }).simulate('click');
    setTimeout(() => {
      expect(Alert.success).toBeCalled();
      done();
    });
  });

  test('Delete user', (done) => {
    const wrapper = shallow(<UserPersonal />);
    mockAdapter.onDelete('user').reply(200);

    wrapper.find({ children: 'Slett meg' }).simulate('click');
    setTimeout(() => {
      expect(userService.token).toBe(null);
      done();
    });
  });

  test('onMount', (done) => {
    const wrapper = shallow(<UserPersonal />);

    setTimeout(() => {
      done();
    });
  });
  test('onMount', (done) => {
    const wrapper = shallow(<UserPersonal />);

    setTimeout(() => {
      done();
    });
  });
  test('failedMount', (done) => {
    userService.token = null;

    const wrapper = shallow(<UserPersonal />);

    setTimeout(() => {
      expect(userService.token).toBe(null);
      done();
    });
  });
});

describe.skip('UserPage component', () => {
  test('On logged out', (done) => {
    const wrapper = shallow(<UserPage />);

    setTimeout(() => {
      expect(
        wrapper.containsMatchingElement(
          // @ts-ignore
          <UserRegister />
        )
      ).toEqual(true);
      expect(
        wrapper.containsMatchingElement(
          // @ts-ignore
          <UserPage />
        )
      ).toEqual(false);
      done();
    });
  });

  test('On logged in', (done) => {
    const wrapper = shallow(<UserPage />);

    userService.token = { id: 0, token: 'token' };

    setTimeout(() => {
      expect(
        wrapper.containsMatchingElement(
          // @ts-ignore
          <UserRegister />
        )
      ).toEqual(false);
      expect(
        wrapper.containsMatchingElement(
          // @ts-ignore
          <UserPersonal />
        )
      ).toEqual(true);
      done();
    });
  });
});
