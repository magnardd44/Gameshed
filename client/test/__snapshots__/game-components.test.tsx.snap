// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddGame component AddGame default 1`] = `
<Container>
  <Card
    title="Legg til ett spill"
  >
    <FormContainer>
      <FormGroup>
        <FormLabel>
          Tittel:
        </FormLabel>
        <FormInput
          id="titleInput"
          onChange={[Function]}
          placeholder="Skriv inn tittel"
          type="text"
          value=""
        />
      </FormGroup>
      <FormGroup>
        <FormLabel>
          Beskrivelse:
        </FormLabel>
        <FormTextarea
          cols={10}
          id="descriptionInput"
          onChange={[Function]}
          placeholder="Skriv inn en beskrivelse av spillet"
          rows={10}
          value=""
        />
      </FormGroup>
      <FormGroup />
    </FormContainer>
    <Linebreak />
    <Row>
      <Column
        width={2}
      >
        <FormLabel>
          Sjanger:
        </FormLabel>
      </Column>
      <Column>
        <Row
          key="0"
        >
          <Column>
            <FormSelect
              id="genreSel"
              key="0"
              onChange={[Function]}
            >
              <option
                hidden={true}
              >
                Velg sjanger her:
              </option>
              <option
                key="0"
                value={1}
              >
                test
              </option>
              <option
                key="1"
                value={2}
              >
                test2
              </option>
            </FormSelect>
          </Column>
        </Row>
        <Linebreak />
      </Column>
    </Row>
    <Row>
      <Column>
        <ButtonSuccess
          onClick={[Function]}
        >
          Legg til flere sjangere (
          2
          )
        </ButtonSuccess>
      </Column>
      <Column />
    </Row>
    <Linebreak />
    <Row>
      <Column
        width={2}
      >
        <FormLabel>
          Platform:
        </FormLabel>
      </Column>
      <Column>
        <Row
          key="0"
        >
          <Column>
            <FormSelect
              id="platformSel"
              key="0"
              onChange={[Function]}
            >
              <option
                hidden={true}
              >
                Velg platform her:
              </option>
              <option
                key="0"
              />
              <option
                key="1"
              />
            </FormSelect>
          </Column>
        </Row>
        <Linebreak />
      </Column>
    </Row>
    <Row>
      <Column>
        <ButtonSuccess
          onClick={[Function]}
        >
          Legg til flere platformer (
          3
          )
        </ButtonSuccess>
      </Column>
      <Column />
    </Row>
  </Card>
  <Card
    title=""
  >
    <Alert />
    <Row>
      <Column>
        <ButtonDanger
          onClick={[Function]}
        >
          Avbryt
        </ButtonDanger>
      </Column>
      <Column>
        <ButtonSuccess
          onClick={[Function]}
        >
          Lagre
        </ButtonSuccess>
      </Column>
    </Row>
  </Card>
</Container>
`;
